### DISCUSSION TITLE
Finding which Bands are Currently in Use

### DISCUSSION THREAD

**Discussion by stellarpower** [gazzasat/huawei-cpe#3] [General]
Hi,

Sorry for a babyish question! I don't know much about WAN-side networking.

Looks like my router is almost identical to what is in the readme (no branding on the outside but 3's locked-down firmware underneath, although, I am using ID, which is on their network anyway). And I see the signal level meters jumping around a bit, which makes me think if maybe it's a similar situation. I think I'm possibly between several towers and it might be switching between them. If I reboot it I get very good speeds immediately after, but over the course of a day, it starts to get very sluggish by comparison.

Tried using [this API package](https://github.com/Salamek/huawei-lte-api) to find out more, but the login flow doesn't appear to be working,so I don't have much more info to go on.

As my bands are currently set to auto, is there a way I can find out what bands it's currently utilising? Or would I need to go to e.g. cellmapper.net and work it out by hand? From the widgets in the webpage I get:

> ENB ID:50066
> CELL ID:12816898
> PCI :488 
> Neighbours :No1:481No2:289 
> MAIN: 3:(15MHz/15MHz) 
> BANDS:AUTO <<<<<
> 5GNR DL:100MHz

And afraid I'm not that sure how to interpret the info beyond higher dBm is probably a good thing!

Thanks!

**Comment by gazzasat** [comment]
Hi,

No worries.  The huawei-lte-api package is something I used before on an older router (B525) as it worked fine but on the newer generation of Huawei devices and especially any that have been 'locked down' by providers the API access has been removed in firmware.

This is the reason you use the javascript injection method which is used in this library... You login to the router admin as normal then apply the script to the page which gets you access to the router modem API in a different way.

You can see that in your info snippet that it says _MAIN: 3:(15MHz/15MHz)_ which shows that when left to AUTO select it is connecting to B3 on your cell with what seems to be aggregation of dual 15MHz carriers.  Is this the combo that gives you good speeds? You can observe and capture the details with good speed and also check what it is when it goes slowly.  

This might be congestion on the same cell tower over the course of the day or maybe it is switching to a different tower / band combo which is less performant.  That is the nature of AUTO band selection.

Using the utility you can filter in/out by band(s) which you might find can stop it switching to a different tower with congestion and poor throughput or to a band (like band say B20) which will give you only slower 5MHz channel but shows a stronger signal and therefore is favoured.

Cellmapper will give you the location of the cell towers and bands serviced in your locality as a start to the detective work. Take some time to measure and record the behaviour and then you can go from there.

Hopefully that gives you something to work on.

**Comment by stellarpower** [comment]
Yeah, I feared it was gonna be more complicated than a simple header or something, but thought maybe the API was handling some of this, as it's apparently supported. Worst-case I could use something like Playwright and drive a headless browser, and some things might be injectable through a browser session. I had read that the Huawei routers were quite good, but I've been a bit disappointed. Better than when I was on Vodafone ADSL because that just went down almost completely every so often and at least this is slow but more inconvenience than unusable. and I've had routers more locked down than this but still, it's not a good number of settings to have. Might still keep an eye out on ebay if another comes up at a good price. And hopefully the networks will put in the effort to keep things improving.

Okay, so when it says Main: that's the same as I would type into the bands box then. I guessed that might be the case, but, the old adage of making an "ass" out of yourself :rofl:. I'll keep an eye on it to start with I guess, and write some notes down, as well as check out CellMapper in more detail. It's always better after a restart so I should be able to compare against that. I wondered if it might be congestion, although I'm pretty nocturnal and it's not necessarily any better at 4AM. Might also be traffic-shaping as even though I'm on an unlimited plan I expect they don't really want me using 300G a month as I am. And then maybe the router is just a bit overloaded, I should work out what the logs are saying.

Thanks a lot for the explanation, appreciate it!

**Reply by gazzasat** [reply]
No problem.  I don't actually use my Huawei any more as shifted over to a Zyxel NR5103E I picked-up from CeX.  The performance is a little better but then again the Zyxel deals better with my external Poynting 4G/5G antenna. It has the band selection built-in to the admin pages.

You are on the right track with measuring first before you start changing things... it just gets confusing and frustrating if you don't get a handle on what the modem does in each situation.

Three UK use 4G bands B1,B3,B20,B28 and supplemental B32 (SDL band is download only - **NEVER** select just this as the filter as the modem can never connect) as well as several 5G NR bands but the Huawei only supports N78 on 5G which you can't filter anyway.   

Currently with 5G non-stand alone (NSA) you will need to be connected to a 4G tower which supports ENDC which 'directs' the modem to a 5G signal which may be on another mast somewhere. You might get one pop-up if you connect can to the anchor mast broadcasting the ENDC signal...

So with AUTO you are effectively using setting 1+3+20+28+32 so you can try some combination of these to try and steer the Huawei onto a different local mast.

Hope that works out.

**Reply by stellarpower** [reply]
Sorry to ask other unrelated questions; I was just looking at an external antenna now I've finally moved to a home I own. The Poynting ones seem to be mentioned quite often and not too bad pricewise on eBay. Did you find it worked well for you? Back when I first ditched my DSL the Hauwei routers seemed to be well-reviewed with some mention that the technology was moving on, but based on my experience and that of others it seems, wouldn't exactly recommend it - the trials of being an early adopter!! So this time I'm cautious to spend more time researching. I may upload my notes if I get a better idea of how it all fits together later and also figured I may fork as it would be nice to add links on the tower/cell IDs to go to cellmapper, OpenCellID, etc. pages - but the latter seem to have less data and the former have locked down their API to an extent, at least it would take more than opening a URL.

**Reply by gazzasat** [reply]
I have one of the Poynting 5G directional (Wideband 2x2) antenna which worked really well for me.  The challenge I had was not signal strength but tower selectivity.

I am in a major city and there are a lot of towers around, many very congested but closer and stronger so an omni antenna or the internal antennas make things worse ... I needed to filter out bands (especially B20 & B3) and point at a paricular 5G ENDC bearing tower to get a stable high speed multi CA connection.

The Poynting 5G antenna I have was not great on B32 (1500MHz) but good enough to bring in too.

I looked at doing tower lookup with the crowd sourced data like Cellmapper but the API is indeed (quite rightly) locked down as hosting such a valuable service is not free! The site needs all the support it can get...

Maybe there are others now like bidb that might work but TBH I now have Gb FTTP and only use the Huawei occasionally as backup.

**Reply by stellarpower** [reply]
Thanks a lot, yeah, similarish situation so been wondering the same. With the thick stone walls it's not surprising that I get bugger all indoors, even just for voice. Inside it seems to lock onto a B3 tower close by (but search doesn't find it or the cell ID on the map), outside it is picking one on a very tall hill with a giant mast a couple of miles away on B1 and getting good signal, so that figures. But I've not worked out where it hands the 5G off to as the tower doesn't seem to list support itself.

Seems the Poynting v3 (confusingly they seem to zero-index the product codes, so, XPOL-2) is directional and offers much better gain. I think I can probably aim it in the right direction but then I may not be close for the handoff tower. Conversely the omndirectional could suffer lots of noise. Guess I could try a sort of Faraday cage to direct it! That said unlike at the previous address, the router is reporting only a single band. There's a B3 just down the road but it doesn't seem to notice it, or at least, Huawei's API is just masking that, perhaps because it's not 5G capable so it gets eliminated from the set.

> hosting such a valuable service is not free! The site needs all the support it can get...

agreed, although, linking to it would permit more people to know it's there. I don't see anything for applying for API keys if you're a bona fide open-source project for non-commercial use, although that wouldn't help much in this scenario as too much to do via simple JS on a different domain. Also either way would be nice if the URLs were stateless, I can't note down a tower for reference later without going back and searching.
